libraries/Filter/NotchFilter.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "NotchFilter.h"
+
+/*
+  initialise filter
+ */
+template <class T>
+void NotchFilter<T>::init(float sample_freq_hz, float center_freq_hz, float bandwidth_hz, float attenuation_dB)
+{
+    float omega = 2.0 * M_PI * center_freq_hz / sample_freq_hz;
+    float octaves = log2f(center_freq_hz  / (center_freq_hz - bandwidth_hz/2)) * 2;
+    float A = powf(10, -attenuation_dB/40);
+    float Q = sqrtf(powf(2, octaves)) / (powf(2,octaves) - 1);
```

Comment:
no need for safe-sqrt instead of sqrtf?

Reply:
we could make it refuse to initialise and act as pass-thru if the params are bad. Using safe_sqrt() would just hide it, but not actally work